File: skypydb/api/collection.py
"""Collection wrapper for vector operations."""

from typing import Any, Dict, Optional, TYPE_CHECKING

from skypydb.api.mixins.vector.collection import (
    AddItemsMixin,
    CollectionInfoMixin,
    DeleteItemsMixin,
    GetItemsMixin,
    QueryItemsMixin,
    UpdateItemsMixin,
)

if TYPE_CHECKING:
    from skypydb.database.vector_db import VectorDatabase


class Collection(
    AddItemsMixin,
    GetItemsMixin,
    QueryItemsMixin,
    UpdateItemsMixin,
    DeleteItemsMixin,
    CollectionInfoMixin,
):
    """Represents a vector collection in the database."""

    def __init__(
        self,
        db: "VectorDatabase",
        name: str,
        metadata: Optional[Dict[str, Any]] = None,
    ):
        """Create a collection wrapper bound to a single `VectorDatabase` instance."""

        self._db = db
        self._name = name
        self._metadata = metadata or {}

File: skypydb-js/src/api/collection.ts
import type { Metadata } from "../types";
import type { VectorDatabase } from "../database/vector_db";
import { CollectionInfoMixin } from "./mixins/vector/collection";

class CollectionBase extends CollectionInfoMixin {}

export class Collection extends CollectionBase {
  protected _db: VectorDatabase;
  protected _name: string;
  protected _metadata: Metadata;

  constructor(db: VectorDatabase, name: string, metadata: Metadata = {}) {
    super();
    this._db = db;
    this._name = name;
    this._metadata = metadata;
  }
}
